pkg/operator/controller/certificate-publisher/publish_certs_test.go

```diff
@@ -139,55 +177,48 @@ func TestDesiredRouterCertsGlobalSecret(t *testing.T) {
 		{
 			description: "no secrets",
 			inputs: testInputs{
-				[]operatorv1.IngressController{ci1},
+				[]operatorv1.IngressController{defaultCI},
 				[]corev1.Secret{},
 			},
 			output: testOutputs{nil},
 		},
 		{
 			description: "missing secret",
 			inputs: testInputs{
-				[]operatorv1.IngressController{ci1, ci2},
-				[]corev1.Secret{s1},
-			},
-			output: testOutputs{
-				&corev1.Secret{
-					Data: map[string][]byte{"dom1": data},
-				},
+				[]operatorv1.IngressController{defaultCI},
+				[]corev1.Secret{s1, s2},
 			},
+			output: testOutputs{nil},
 		},
 		{
 			description: "extra secret",
 			inputs: testInputs{
-				[]operatorv1.IngressController{ci2},
-				[]corev1.Secret{s1, s2},
+				[]operatorv1.IngressController{defaultCI, ci2},
+				[]corev1.Secret{s1, defaultCert, s2},
 			},
 			output: testOutputs{
 				&corev1.Secret{
-					Data: map[string][]byte{"dom2": data},
+					Data: map[string][]byte{"apps.my.devcluster.openshift.com": defaultCertData},
 				},
 			},
 		},
 		{
 			description: "perfect match",
 			inputs: testInputs{
-				[]operatorv1.IngressController{ci1, ci2},
-				[]corev1.Secret{s1, s2},
+				[]operatorv1.IngressController{defaultCI, ci1, ci2},
+				[]corev1.Secret{defaultCert, s1, s2},
 			},
 			output: testOutputs{
 				&corev1.Secret{
-					Data: map[string][]byte{
-						"dom1": data,
-						"dom2": data,
-					},
+					Data: map[string][]byte{"apps.my.devcluster.openshift.com": defaultCertData},
 				},
 			},
 		},
 	}
 
 	for _, tc := range testCases {
 		expected := tc.output.secret
-		actual, err := desiredRouterCertsGlobalSecret(tc.inputs.secrets, tc.inputs.ingresses, "openshift-ingress")
+		actual, err := desiredRouterCertsGlobalSecret(tc.inputs.secrets, tc.inputs.ingresses, "openshift-ingress", "apps.my.devcluster.openshift.com")
 		if err != nil {
 			t.Errorf("failed to get desired router-ca global secret: %v", err)
 			continue
```

Comment:
I know this line wasn't modified but I changed the overall structure to use subtests so that I could fiddle with the expected inputs/output and just invoke a particular test case to observe deliberation changes to the test conditions didn't allow the test to pass. But the change also helped convince me that the early `continue` statements didn't mask any other assertions that should/could have fired.

```suggestion
	for _, tc := range testCases {
		t.Run(tc.description, func(t *testing.T) {
			expected := tc.output.secret
			actual, err := desiredRouterCertsGlobalSecret(tc.inputs.secrets, tc.inputs.ingresses, "openshift-ingress", "apps.my.devcluster.openshift.com")
			if err != nil {
				t.Errorf("failed to get desired router-ca global secret: %v", err)
			}
			if expected == nil || actual == nil {
				if expected != nil {
					t.Errorf("expected %v, got nil", expected)
				}
				if actual != nil {
					t.Errorf("expected nil, got %v", actual)
				}
			}

			if expected != nil && actual != nil {
				if !routerCertsSecretsEqual(expected, actual) {
					t.Errorf("expected %v, got %v", expected, actual)
				}
			}
		})
	}
```

Reply:
Are you suggesting that I change `TestDesiredRouterCertsGlobalSecret` to use `t.Run`?

Reply:
I went ahead and added a commit with your suggestion.